### ISSUE TITLE
[Feedback]Multiple errors in the meat of the tutorial - giving up

### ISSUE THREAD

**Issue by codeedog** [aws-amplify/docs#3140]
**Page**: [`/start/getting-started/nextsteps/q/integration/angular`](https://docs.amplify.aws/start/getting-started/nextsteps/q/integration/angular)

**Feedback**:

I tried following along with your tutorial, but there are too many errors and I don't feel like continuing through with figuring out how to fix them. Normally, I try to help improve the projects I'm interested in using, however, I already had to fight through a bug on the previous [page](https://docs.amplify.aws/start/getting-started/setup/q/integration/angular) which I encountered when running `amplify init` and was able to work around by using `amplify add env`, but only after stumbling upon a link in a comment in this StackOverflow [issue](https://stackoverflow.com/questions/67068703/typeerror-cannot-read-property-projectpath-of-undefined-when-create-amplify) filed by someone else.

I'm more interested in understanding how to create a backend in REST for my Angular front end. I want the backend to be NestJs compliant, which means Typescript. Some of the errors below are Angular typescript errors and are in your provided libraries. Other of the errors are code that I copied and pasted from your tutorial. It's possible that I made a mistake, but since GraphQL is not of my interest and there are many errors, I'm giving up.

Which, tbh, is a shame, because I would like to understand GQL better, and I will, down the road.

You should fix your libraries, at a minimum, so that they are typescript compliant or provide the `.d.ts` type extensions. The fact that you don't leads me to believe that the rest of this tutorial has a similar level of inattention and is not worth my time.

I apologize for being blunt. It is not intended as disparagement. Rather, it is intended to let you know that I'm new to AWS but very much not new to programming and am judging your company by the quality its product including its teaching products.

````
Error: node_modules/@aws-amplify/api-graphql/lib-esm/types/index.d.ts:1:30 - error TS7016: Could not find a declaration file for module 'graphql/error/GraphQLError'. '/Users/andrewphilips/Documents/Websites/test-aws/aws-ng-tutorial/amplify-app/node_modules/graphql/error/GraphQLError.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/graphql` if it exists or add a new declaration (.d.ts) file containing `declare module 'graphql/error/GraphQLError';`

1 import { GraphQLError } from 'graphql/error/GraphQLError';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/@aws-amplify/api-graphql/lib-esm/types/index.d.ts:2:30 - error TS7016: Could not find a declaration file for module 'graphql/language/ast'. '/Users/andrewphilips/Documents/Websites/test-aws/aws-ng-tutorial/amplify-app/node_modules/graphql/language/ast.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/graphql` if it exists or add a new declaration (.d.ts) file containing `declare module 'graphql/language/ast';`

2 import { DocumentNode } from 'graphql/language/ast';
                               ~~~~~~~~~~~~~~~~~~~~~~


Error: src/app/app.component.html:2:28 - error NG8002: Can't bind to 'formGroup' since it isn't a known property of 'form'.

2   <form autocomplete="off" [formGroup]="createForm" (ngSubmit)="onCreate(createForm.value)">
                             ~~~~~~~~~~~~~~~~~~~~~~~~

  src/app/app.component.ts:8:16
    8   templateUrl: './app.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component AppComponent.


Error: src/app/app.component.ts:14:10 - error TS2564: Property 'createForm' has no initializer and is not definitely assigned in the constructor.

14   public createForm: FormGroup;
            ~~~~~~~~~~


Error: src/app/app.component.ts:16:3 - error TS2564: Property 'restaurants' has no initializer and is not definitely assigned in the constructor.

16   restaurants: Array<Restaurant>;
     ~~~~~~~~~~~


Error: src/app/app.component.ts:29:7 - error TS2322: Type '({ __typename: "Restaurant"; id: string; name: string; description: string; city: string; createdAt: string; updatedAt: string; } | null)[] | null | undefined' is not assignable to type 'Restaurant[]'.
  Type 'undefined' is not assignable to type 'Restaurant[]'.

29       this.restaurants = event.items;
         ~~~~~~~~~~~~~~~~


mError: Entry module not found: Error: Can't resolve '/Users/andrewphilips/Documents/Websites/test-aws/aws-ng-tutorial/amplify-app/src/app/app.component.css' in '/Users/andrewphilips/Documents/Websites/test-aws/aws-ng-tutorial/amplify-app'
````

**Comment by medelman17** [comment]
Hi @codeedog! Blunt or not, we appreciate the feedback. We're definitely interested in improving the libraries and related documentation. It seems like there's a lot to unpack, here. Would you be interested in jumping on a call next week to discuss? I want to better understand the specific pain points you encountered.

**Comment by codeedog** [comment]
Happy to jump on a call. PM or email to figure out availability?

**Comment by medelman17** [comment]
Sorry for the delayed response. Last week got away from me. Hit me up on twitter: https://twitter.com/edelman215 DMs are open!

**Comment by wlee221** [comment]
Hello @codeedog, we much appreciate the feedback. The Angular documentation gap has widened especially since Angular 12, which uses `strict` typescript mode. We just pushed in #3606, which strictly types the variables we have in  the tutorial so it's compliant with latest Angular. 

Please see the callouts as well (grey boxes) in the updated [documentation](https://docs.amplify.aws/start/getting-started/data-model/q/integration/angular/#connect-frontend-to-api) where we provide gists to declaration file you can use to workaround the `graphql` and `awsExports` type issues. 

We'll go ahead and close this issue as the errors you have provided have been addressed from #3606, but please let us know if you see any other Angular blocker or any concern in general. We'll work to bridge the Angular gap we have!
